chia/data_layer/data_layer_wallet.py

```diff
@@ -521,6 +542,49 @@ async def create_update_state_spend(
         )
         return txs
 
+    async def generate_signed_transaction(
+        self,
+        amounts: List[uint64],
+        puzzle_hashes: List[bytes32],
+        fee: uint64 = uint64(0),
+        coins: Set[Coin] = set(),
+        memos: Optional[List[List[bytes]]] = None,  # ignored
+        coin_announcements_to_consume: Optional[Set[Announcement]] = None,
+        puzzle_announcements_to_consume: Optional[Set[Announcement]] = None,
+        ignore_max_send_amount: bool = False,  # ignored
+        # This wallet only
+        launcher_id: Optional[bytes32] = None,
+        new_root_hash: Optional[bytes32] = None,
+        sign: bool = True,  # This only prevent signing of THIS wallet's part of the tx (fee will still be signed)
+    ) -> List[TransactionRecord]:
+        # Figure out the launcher ID
+        if len(coins) == 0:
+            if launcher_id is None:
+                raise ValueError("Not enough info to know which DL coin to send")
+        else:
+            if len(coins) != 1:
+                raise ValueError("The wallet can only send one DL coin at a time")
```

Comment:
How come we're limited to only sending one DL coin at a time? I'm not sure of the possible use cases - would we ever have a scenario of wanting to do bulk updates on multiple DLs?

Reply:
You're right that we could do that, however we'd have to work out some complex-ish logic for making sure the coins are dependent on each other and don't all try to spend the same fee coin etc.  Since that didn't already exist and wasn't a current requirement, I decided to just restrict the user to one DL transaction at a time.